app/services/vendor_api/generate_test_applications_for_provider.rb

```diff
@@ -12,16 +12,45 @@ def call(provider:, courses_per_application:, count:, for_ratified_courses: fals
 
   private
 
-    def course_list_for_provider(provider, for_ratified_courses)
+    def course_list_for_provider(provider, for_ratified_courses, mixed_application)
       if for_ratified_courses
-        GetCoursesRatifiedByProvider.call(provider: provider)
+        courses_ratified_by_provider(provider)
+      elsif mixed_application
+        [
+          courses_run_by_provider(provider).first,
+          courses_ratified_by_provider(provider).first,
+          course_run_by_test_provider,
+        ].compact
       else
-        Course.current_cycle
-              .open_on_apply
-              .joins(:course_options)
-              .distinct
-              .where(provider: provider)
+        courses_run_by_provider(provider)
       end
     end
+
+    def courses_ratified_by_provider(provider)
+      GetCoursesRatifiedByProvider.call(provider: provider)
+    end
+
+    def courses_run_by_provider(provider)
+      Course.current_cycle
+            .open_on_apply
+            .joins(:course_options)
+            .distinct
+            .where(provider: provider)
+    end
+
+    def course_run_by_test_provider
+      test_provider = Provider.default_scoped.find_or_create_by(name: 'Test Provider', code: 'TEST')
```

Comment:
I thought we had a test provider for everyone, I prefer going with one test provider per provider, but wanted to double check after our conversation. Can we reuse `GenerateFakeProvider` to return the existing one with the courses required?